sql/ha_partition.cc

```diff
@@ -5663,6 +5663,19 @@ int ha_partition::read_range_next()
   DBUG_RETURN(handle_unordered_next(table->record[0], eq_range));
 }
 
+bool ha_partition::has_gap_locks()
```

Comment:
Let's make a note somewhere (maybe in the 8.0 blueprint), that when legacy partitioning is dead, this method should move from handler (per-table property) to handlerton (per-SE property), which it fundamentally is, if not ha_partition